src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.cs

```diff
@@ -265,6 +265,12 @@ public string SpecialBuild
         /// </summary>
         public static FileVersionInfo GetVersionInfo(string fileName)
         {
+            // Check if fileName is a full path. Relative paths can cause confusion if the local file has the .dll extension, 
+            // as .dll search paths can take over & look for system .dll's in that case.
+            if (!Path.IsPathFullyQualified(fileName))
```

Comment:
If somebody calls it with fileName == null, it will throw ArgumentNullException now. I think it was throwing FileNotFoundException before.

Reply:
Would you suggest try/catching this? Or special casing null?

Reply:
I would special case null.

Reply:
(Also, add a test for it.)